ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/precompiles/BLS12PairingPrecompiledContract.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+package org.hyperledger.besu.ethereum.mainnet.precompiles;
+
+import org.hyperledger.besu.ethereum.core.Gas;
+import org.hyperledger.besu.nativelib.bls12_381.LibEthPairings;
+
+import org.apache.tuweni.bytes.Bytes;
+
+public class BLS12PairingPrecompiledContract extends AbstractBLS12PrecompiledContract {
+
+  public BLS12PairingPrecompiledContract() {
+    super("BLS12_PAIRING", LibEthPairings.BLS12_PAIR_OPERATION_RAW_VALUE);
+  }
+
+  @Override
+  public Gas gasRequirement(final Bytes input) {
+    final int k = input.size() / 384;
+    return Gas.of(23_000L * k + 115_000);
```

Comment:
why considering 23_000 as large?

Reply:
Integer overflow.  LGTM.com will complain that we are doing integer math then casting it to a long and that there may be a surprising loss in precision.  By starting out with a long constant the whole string of math is done at Long width.  It would take an amazingly large `k` value to overflow a 32 bit int, but it keeps the automated checkers happy.

Reply:
👍